units_godot/global.gd

```diff
@@ -1,19 +1,52 @@
 extends Node
 
-const GameState = preload("res://lib/game_state.gd")
-const Resources = preload("res://lib/resources.gd")
-const AiGameManager = preload("res://lib/ai_game_manager.gd")
-onready var units = get_node("/root/UnitKinds").units
+const AiGameManager = preload('res://lib/ai_game_manager.gd')
+const Action = preload('res://native/action.gdns')
+const Colour = preload('res://native/colour.gdns')
+const GameRules = preload('res://native/game_rules.gdns')
+const GameState = preload('res://native/game_state.gdns')
 
-# jim: We initialize man to a good value here so that we can
-# use the Play Scene feature in Godot.
-onready var man = AiGameManager.new(units) # Game manager
+var rules
+var unit_kinds
+var man
+
+# So our naming convention here is apparently:
+# Thing: Thin wrapper over C++ pointer.;
+# Thing_: Object that provides gdscript-level conveniences.
+# Thing__: Class for Thing_.
+class Colour__:
```

Comment:
Why is this declared here, rather than in lib/colour.gd or something? Will we need/want similar Thing__-classes for all the other C++ classes, on top of all the Things?

Reply:
Declared here to make it available globally without the overhead (quite substantial in gdscript) of separate files and explicit imports. We won't need Thing__s for the most part, but Colour_ was special because I wanted three instances of Colour that we use again and again, rather than having to instantiate a new Colour every time we use one. Ideally we could just use Colour.RED instead of doing this, but gdscript has no static variables.